comp110-worksheet-6/OxoBoard.cs

```diff
@@ -10,52 +10,165 @@ public enum Mark { None, O, X };
 
 	public class OxoBoard
 	{
+
+        public int width;
+        public int height;
+
+        public Mark[,] board;
+
+        public string[] wins;
+
 		// Constructor. Perform any necessary data initialisation here.
 		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
 		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            width = 3;   //Set width and height of board, with a default value of 3x3
+            height = 3;  //
+            board = new Mark[width, height]; //Initialise the board
+
+            wins = GenerateWins(3);
+
+            // Set all the board spaces to None
+            for (int i = 0; i < width; i++)
+            {
+                for (int j = 0; j < height; j++)
+                {
+                    board[i, j] = Mark.None;
+                }
+            }
 		}
 
+        public OxoBoard(int size)
+        {
+            width = height = size;
+            board = new Mark[width, height]; //Initialise the board
+
+            wins = GenerateWins(size);
+
+            // Set all the board spaces to None
+            for (int i = 0; i < width; i++)
+            {
+                for (int j = 0; j < height; j++)
+                {
+                    board[i, j] = Mark.None;
+                }
+            }
+        }
+
+        public string[] GenerateWins(int size)
+        {//generate all the win states. As this board is square, they will be horizontal (1 per line), vertical (1 per column) and 2 diagonals
+            List<string> ret = new List<string>();
+            for (int i = 0; i < size; i++)
+            {
+                string row = "";
+                for (int j = 0; j < size; j++)
+                {
+                    row += i.ToString() + j;
```

Comment:
This will fail for anything larger than a 10x10 board. Better to use a list of ints rather than a string?